### ISSUE TITLE
Rules endpoint 500 without auth header (patched single process mode)

### ISSUE THREAD

**Issue by robx** [cortexproject/cortex#2746]
Talking to single-process cortex that's been [patched](https://github.com/hoenn/cortex/commit/9eec635af23a71aa36dfe607afe599cd4cf45a8b) to include the ruler, I see a 500 error when GETting `/api/prom/api/v1/rules`:

```
$ curl -i http://localhost:9009/api/prom/api/v1/rules; echo
HTTP/1.1 500 Internal Server Error
Date: Thu, 18 Jun 2020 09:11:45 GMT
Content-Length: 89
Content-Type: text/plain; charset=utf-8

{"status":"error","data":null,"errorType":"server_error","error":"no valid org id found"}
```

It seems this would better be a 401 (or maybe 400).

The log output:

```
cortex_1         | level=error ts=2020-06-18T09:11:45.45160183Z caller=api.go:126 org_id=fake traceID=4445bbca909df6 msg="error extracting org id from context" err="no org id"
cortex_1         | level=warn ts=2020-06-18T09:11:45.451764394Z caller=logging.go:62 traceID=4445bbca909df6 msg="GET /api/prom/api/v1/rules (500) 144.429µs Response: \"{\\\"status\\\":\\\"error\\\",\\\"data\\\":null,\\\"errorType\\\":\\\"server_error\\\",\\\"error\\\":\\\"no valid org id found\\\"}\" ws: false; Accept: */*; User-Agent: curl/7.54.0; "
```

If the `X-Scope-OrgId` header is passed, the request works fine:

```
$ curl -H "X-Scope-OrgId: 1" -i http://localhost:9009/api/prom/api/v1/rules; echo
HTTP/1.1 200 OK
Content-Type: application/json
Date: Thu, 18 Jun 2020 09:17:18 GMT
Content-Length: 67

{"status":"success","data":{"groups":[]},"errorType":"","error":""}
```

so I'm hoping the setup is close enough to a vanilla cortex+ruler that the same occurs there. I don't have that set up yet to verify but happy to dig in further.

**Comment by gotjosh** [comment]
@pracucci can you please unstale and assign to me?

**Comment by pracucci** [comment]
> @pracucci can you please unstale and assign to me?

@gotjosh Sure, done.

**Comment by pracucci** [comment]
The PR #3343 should fix it.

**Comment by pracucci** [comment]
Fixed by #3343
